tests/test_data_types.py

```diff
@@ -241,6 +241,29 @@ def test_matplotlib_image():
     img = wandb.Image(plt)
     assert img._image.width == 640
 
+def test_matplotlib_image_with_multiple_axes():
+    '''Ensures that wandb.Image constructor can accept a pyplot reference in
+    which the figure has multiple axes. Importantly, there is no requirement
+    that any of the axes have plotted data.
+    '''
+    data = [1,2,3]
+    total_plot_count = 3 # results in 3 plots
+
+    # Attempts to create a figure containing up to `total_plot_count` axes and
+    # optionally adds data to each axes in a permutation-style loop.
+    for num_plots in range(1, total_plot_count + 1):
+        for permutation in range(2 ** num_plots):
+            has_data = [permutation & (1 << i) > 0 for i in range(num_plots)]
+            fig, ax = plt.subplots(num_plots)
+            if num_plots == 1:
+                if has_data[0]:
+                    ax.plot(data)
+            else:
+                for plot_id in range(num_plots):
+                    if has_data[plot_id]:
+                        ax[plot_id].plot(data)
+            
+            wandb.Image(plt) # this should not error.
```

Comment:
wandb.Plotly is the other datatype that wraps matplotlib figures. Can you make sure this test covers that too?

Reply:
Hey @adrnswanberg, great call. I started looking into this and realized that the currently implementation of wandb.Plotly does not accept a matplotlib figure (despite what the docs say). In fact, it only accepts a Plotly figure. Just running some tests now, but the followup commit will have tests and support for such implicit conversion.

Reply:
Followup from verbal conversation with @adrnswanberg... cc @raubitsj 

Hi Jeff,
Adrian and I just met to discuss this change and wanted to invite you to weigh in. Here are some of the key points:
1. The initial bug was caused because the `util.ensure_matplotlib_figure` method was calling `<pyplot>.gca().has_data()` which fails when the last axes of a plot does not contain data. We made the call to remove this constraint and allow the use to log figures even when data is absent. 
2. As a followup, Adrian pointed out that we should also test the same functionality of the `wandb.Plotly` constructor. Digging in, I found that this constructor does not actually accept pyplot objects (despite the docs). See line 1287 of data_types.py. In response I modified the constructor to make this implicit conversion
3. After speaking with Adrien, we felt it might be questionable to introduce this functionility given that it was not already present, and instead, simply have the test validate that the static `Plotly.make_plot_media` method can accept the pyplot object. This function seems to be the key function used in the recursive serialization of the logged dictionaries. As a result, we would want to remove the documentation stating the the Plotly constructor can accept matplotlib objects.

We would be happy to get your thoughts on this approach, especially as it would be "removing" stated, but not implemented, functionality.

Thanks,
Tim

Reply:
I think we should make wandb.Plotly accept plotly objects.  It's not a breaking change as it will just add new functionality / fix something that is obviously broken.  I may be confusing plotly for pyplot, but my same logic applies.  Adding this won't break anyones code more than it's already broken.

Reply:
^ I'm ok with this. I hadn't considered closely enough that we do have documentation stating this is supposed to work.

Reply:
Great! Thanks folks - will move forward with the conversion approach.